### PR TITLE
Add "PasswordMustChange" to New-DbaLogin command

### PR DESCRIPTION
**Pull request by ClaudioESSilva**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Make possible to specify that login being created need to change password at next login

### Approach

When `-PasswordMustChange` is specified, we will enforce the `PasswordExpirationEnabled` and `PasswordPolicyEnforced` to true as they are required for the must change flag be turn on.

### Commands to test
New-DbaLogin with `-PasswordMustChange`

### Screenshots

image

### Learning
As `MustChangePassword` property of the login is just a `get`, we need to use the [ChangePassword](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.login.changepassword?view=sql-smo-160#Microsoft_SqlServer_Management_Smo_Login_ChangePassword_System_Security_SecureString_System_Boolean_System_Boolean_) method

### CODE REVIEW

File: functions/New-DbaLogin.ps1 [hunk 1]
```diff
@@ -493,12 +508,31 @@ function New-DbaLogin {
...
+
                         #Display results
                         # If we ever used T-SQL, the smo is some times not up to date and should be refreshed
                         if ($usedTsql) {
                             $server.Logins.Refresh()
                         }
-                        Get-DbaLogin -SqlInstance $server -Login $loginName
+
+                        if ($loginType -eq 'SqlLogin') {
+                            Get-DbaLogin -SqlInstance $server -Login $loginName | Select-Object ComputerName, InstanceName, SqlInstance, Name, LoginType, CreateDate, LastLogin, HasAccess, IsLocked, IsDisabled, MustChangePassword
```
Comment (nvarscar) [inline functions/New-DbaLogin.ps1:532]: We should return a Login object in all cases. Select-Object would create a custom object instead. Did you want to use Select-DefaultView

File: functions/New-DbaLogin.ps1 [hunk 2]
```diff
@@ -493,12 +508,31 @@ function New-DbaLogin {
                                 }
                             }
                         }
+
+                        #Process the MustChangePassword property
+                        if ($currentPasswordMustChange -ne $newLogin.MustChangePassword) {
+                            try {
+                                $newLogin.ChangePassword($SecurePassword, $true, $true)
```
Comment (nvarscar) [inline functions/New-DbaLogin.ps1:515]: If it requires password change, then we need to validate that the password is specified when the MustChangePassword 
 parameter is used.

File: functions/New-DbaLogin.ps1 (hunk 1 above)
Comment (ClaudioESSilva) [reply functions/New-DbaLogin.ps1]: Excellent catch! 
Changed. Thanks.

File: functions/New-DbaLogin.ps1 (hunk 2 above)
Comment (ClaudioESSilva) [reply functions/New-DbaLogin.ps1:520]: Added validation on the top of the function.
Thanks!

File: functions/New-DbaLogin.ps1 (hunk 1 above)
Comment (nvarscar) [reply functions/New-DbaLogin.ps1]: Not a problem. I'm still wondering though why SqlLogin needs a special treatment here.

File: functions/New-DbaLogin.ps1 (hunk 2 above)
Comment (nvarscar) [reply functions/New-DbaLogin.ps1:520]: Looks good!

File: functions/New-DbaLogin.ps1 (hunk 1 above)
Comment (ClaudioESSilva) [reply functions/New-DbaLogin.ps1]: It's just my suggestion and how I would like it to have it. 

Windows login can't be created with must change pwd flag. Thus, don't make sense to output. 
However, if we are creating one (or more) SQL login and we specify the flag I find it useful to know if it needs to change the password. I see it like we have the isDisabled.

Comment (wsmelton) [reply functions/New-DbaLogin.ps1]: I see this handled better by `Get-DbaLogin` than with the new function. Just add this logic to the Get command.

Comment (wsmelton) [reply functions/New-DbaLogin.ps1]: The same reason could be said on the output itself of `Get-DbaLogin`..so no real reason to not just do it there.

Comment (ClaudioESSilva) [reply functions/New-DbaLogin.ps1]: - Rollback the change of the output on the New-DbaLogin
- Added MustChangePassword to the default output of Get-DbaLogin
- Took the chance to add a new switch parameter -MustChangePassword that will just output logins that need to change password.

Do you agree?

**Comment by potatoqualitee** [comment]
Thanks to everyone who helped review and thank you @ClaudioESSilva for adding this functionality 🙇

**Pull request merged**